src/check_pack.c

```diff
@@ -51,21 +51,22 @@ static int ck_max_msg_size = 0;
  * Problems were seen in the wild with up to 4 GB reallocations.
  */
 
-void check_set_max_msg_size(int max_msg_size) {
-    if (ck_max_msg_size <= 0) {
-        char *env = getenv("CK_MAX_MSG_SIZE");
-        if (env)
-            ck_max_msg_size = atoi(env);
-        if (ck_max_msg_size <= 0)
-            ck_max_msg_size = max_msg_size;
-        if (ck_max_msg_size <= 0)
-            ck_max_msg_size = DEFAULT_MAX_MSG_SIZE;
-    }
+void check_set_max_msg_size(size_t max_msg_size)
+{
+    ck_max_msg_size = 0;
+    char *env = getenv("CK_MAX_MSG_SIZE");
+    if (env)
+        ck_max_msg_size = (size_t)strtoul(env, NULL, 10); // Cast in case size_t != unsigned long.
+    if (ck_max_msg_size == 0)
+        ck_max_msg_size = max_msg_size;
+    if (ck_max_msg_size == 0)
+        ck_max_msg_size = DEFAULT_MAX_MSG_SIZE;
 }
 
-static int get_max_msg_size(void) {
-    if (ck_max_msg_size <= 0)
-        check_set_max_msg_size(0);
+static size_t get_max_msg_size(void)
+{
+    if (ck_max_msg_size == 0)      // If check_set_max_msg_size was not called,
```

Comment:
Should not the getter be making the decision as to if the environment variable exists or not, and let the setter only set the value passed? This way, if the environment variable changed since when the setter was called and the getter was called the correct value would be returned? e.g.:
```
void check_set_max_msg_size(size_t max_msg_size)
 {
     ck_max_msg_size = max_msg_size;
 }

void get_max_msg_size()
{
    size_t value = 0;

    char *env = getenv("CK_MAX_MSG_SIZE");
     if (env)
         value = (size_t)strtoul(env, NULL, 10); // Cast in case size_t != unsigned long.
     if(value == 0)
         value = ck_max_msg_size;
     if (value == 0)
         value = DEFAULT_MAX_MSG_SIZE;

   return value;
}
```

Reply:
Changing the environment variable at run time seems strange, but I guess it could happen. Also, I guess performance isn't an issue, as this is only used when an assertion fails...